File: Theory/Substring/Substring.java
import java.util.Scanner;

public class Substring {
    public static void main(String[] args) {
        String firstString;
        String secondString;

        Scanner input = new Scanner(System.in);

        firstString = input.next();
        secondString = input.next();

        // Contains returns either true or false.
        // If you need the index, you should use indexOf().

        if (firstString.contains(secondString)) {
            System.out.println("Found");
        } else {
            System.out.println("Not found");
        }

        input.close(); // Once closed, you cannot take anymore inputs.
    }
}

File: Theory/Substring/Substring.py
# Python implementation of finding a substring within the string.

first_string = input("Enter a string: ")
second_string = input("Enter the substring to search for: ")

# the find(string_name) returns the first index where the substring is found and -1 if nothing is found.


# The code below just checks if the substring exists or not

if first_string.find(second_string) > -1:
    print("Found")
else:
    print("Not found")